### ISSUE TITLE
enkfgdas_select_obs (eobs) issue with chgrp file obs_input.tar that referenced to an empty obsinput.ensmean file

### ISSUE THREAD

**Issue by lgannoaa** [NOAA-EMC/global-workflow#757]
This bug is also exist in WCOSS2 production GFS job and WCOSS production GFS.
Log file /lfs/h1/ops/prod/output/20220426/enkfgdas_select_obs_00.o2361533 has:
269 + chgrp rstprod obs_input.0001 obs_input.0002 obs_input.0003 obs_input.0005 obs_input.0006 obs_input.0009 obs_input.0010 obs_input.0011 obs_input.0017 obs_input.0018 obs_input.0029 obs_input.0030 obs_input.0041 obs_input.0057 obs_input.0060 obs_input.0061 obs_input.0062 obs_input.0063 obs_input.0066 obs_input.0068 obs_input.0069 obs_input.0070 obs_input.0072 obs_input.0073 obs_input.0076 obs_input.0088 obs_input.0089 obs_input.0090 obs_input.0094 obs_input.0095 obs_input.0098 obs_input.0099 obs_input.0100 obs_input.0101 obs_input.0102 obs_input.common obs_input.tar
chgrp: cannot dereference 'obs_input.tar': No such file or directory
Similar behavior can be found in WCOSS1 NCO GFS production:
/gpfs/dell1/nco/ops/com/output/prod/today/gdas_enkf_select_obs_00.o90805814
**Expected behavior**

chgrp should not have issue in exglobal_atmos_analysis.sh
**Current behavior**

chgrp: cannot dereference 'obs_input.tar': No such file or directory
**Machines affected**

WCOSS2 NCO production GFS gfs.v16.2.0.
WCOSS1 NCO production GFS gfs.v16.1.7

**Detailed Description**

The chgrp display issue message "cannot dereference" when script $HOMEgfs/scripts/exglobal_atmos_analysis.sh try to chgrp file obs_input.tar in DATA directory. File obs_input.tar is a softlink to $COMOUT/gdas.t00z.obsinput.ensmean. This file has not yet been written at the time this script issue chgrp commend.

Instead of chgrp all obs_input.* files in DATA, script exglobal_atmos_analysis.sh should be modified to not chgrp obs_input.tar in the following code session:
if [ $RUN_SELECT = "YES" ]; then
  echo $(date) START tar obs_input >&2
  [[ -s obsinput.tar ]] && rm obsinput.tar
  $NLN $SELECT_OBS obsinput.tar
  ${CHGRP_CMD} obs_input.*              ##################### issue found here 
  tar -cvf obsinput.tar obs_input.*
  chmod 750 $SELECT_OBS
  ${CHGRP_CMD} $SELECT_OBS
  rm obsinput.tar
  echo $(date) END tar obs_input >&2
fi

**Comment by RussTreadon-NOAA** [comment]
@lgannoaa , thank you for finding the same bug which you reported last summer.  NOAA-EMC/GSI issue [#180](https://github.com/NOAA-EMC/GSI/issues/180) was created at that time to document this bug and its resolution.  @MichaelLueken-NOAA merged the fix into develop at [17cddf0](https://github.com/NOAA-EMC/GSI/commit/17cddf0ef79b19e01d55800cd1d3864cc4e0e99e).   

I added the fix to a working copy of `exgdas_enkf_select_obs.sh` on WCOSS2.   A rerun of 2022042612 enkfgdas_select_obs confirmed that the fix still works.   The message 

`chgrp: cannot dereference 'obs_input.tar': No such file or directory
`

is not in the rerun job log file.

**Comment by RussTreadon-NOAA** [comment]
@MichaelLueken-NOAA , the modifed `enkfgdas_select_obs.sh` is on Cactus in `/lfs/h2/emc/da/noscrub/Russ.Treadon/git/global-workflow/feature-ops-wcoss2/sorc/gsi.fd/scripts`.   I also updated the `rm` before the link.   The `rm` isn't really necessary since the operational job runs in a fresh directory.    Here is a `git diff` of the file

```
 # ObsInput file from ensemble mean
-rm -f obs_input*
-$NLN $SELECT_OBS obs_input.tar
+rm -f obs*input*
+$NLN $SELECT_OBS obsinput.tar
 
 # Whether to save or skip obs
```

**Comment by MichaelLueken** [comment]
Thank you @lgannoaa and @RussTreadon-NOAA for bringing this to my attention.  I have made the change to `exgdas_enkf_select_obs.sh`, pushed the updates to `release/gfsda.v16.2.0` (at [694655e](https://github.com/NOAA-EMC/GSI/commit/694655e8cb0ea05c8b5435ce37aa2b2ee0396a97)) and `feature/gfsda.v16.1.5_wcoss2_port` (at [bb5d098](https://github.com/NOAA-EMC/GSI/commit/bb5d098757f2e1620855ae15dcaedcb6a808223b)), and recreated tags gfsda.v16.2.0 and gfsda.v16.2.0.1, respectively.

Tagging @KateFriedman-NOAA to let her know about the updated tags associated with this fix.

**Comment by KateFriedman-NOAA** [comment]
> pushed the updates to release/gfsda.v16.2.0 (at [694655e](https://github.com/NOAA-EMC/GSI/commit/694655e8cb0ea05c8b5435ce37aa2b2ee0396a97)) and feature/gfsda.v16.1.5_wcoss2_port (at [bb5d098](https://github.com/NOAA-EMC/GSI/commit/bb5d098757f2e1620855ae15dcaedcb6a808223b)), and recreated tags gfsda.v16.2.0 and gfsda.v16.2.0.1, respectively

@MichaelLueken-NOAA Thanks for making this change and recutting those tags! The fix will get into WCOSS2 ops with the next updated install (Wei just did one yesterday). Thanks @lgannoaa for reporting this and thanks @RussTreadon-NOAA for making the update to the `exgdas_enkf_select_obs.sh` script.

**Comment by arunchawla-NOAA** [comment]
@lgannoaa thank you for your diligence in finding this !

**Comment by lgannoaa** [comment]
This solution /lfs/h2/emc/da/noscrub/Russ.Treadon/git/global-workflow/feature-ops-wcoss2/sorc/gsi.fd/scripts/exgdas_enkf_select_obs.sh was tested in today's ecflow development realtime parallel. 
A check on testing result shown the bug is no longer exist:
Lin.Gan@cdecflow02:/lfs/h2/emc/ptmp/Lin.Gan/ecfops/com/output/prod/today> ll enkfgdas_select_obs_*.o*                                   
-rw-r--r-- 1 Lin.Gan emc 906K Apr 27 14:31 enkfgdas_select_obs_00.o2449710
-rw-r--r-- 1 Lin.Gan emc 907K Apr 27 17:07 enkfgdas_select_obs_06.o2457067
-rw-r--r-- 1 Lin.Gan emc 903K Apr 27 18:43 enkfgdas_select_obs_12.o2461039
-rw-r--r-- 1 Lin.Gan emc 902K Apr 28 00:06 enkfgdas_select_obs_18.o2477999
Lin.Gan@cdecflow02:/lfs/h2/emc/ptmp/Lin.Gan/ecfops/com/output/prod/today> grep "No such file or directory" enkfgdas_select_obs_*.o*|wc -l
0

**Comment by RussTreadon-NOAA** [comment]
Good that the real-time test confirmed the single test result.   The error message is no longer present.  The fix documented in GSI issue [#180](https://github.com/NOAA-EMC/GSI/issues/180) should have been incorporated into the WCOSS2 GFS DA port earlier in the transition.

**Comment by WalterKolczynski-NOAA** [comment]
This should be fixed in develop as part of #969, as now that `set -e` is on, the errors would block the jobs so we now check the links exist first.

I think this can be closed now?

**Comment by lgannoaa** [comment]
Closing the ticket.
